app/common/model/src/main/java/io/syndesis/common/model/WithRevision.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.common.model;
+
+import java.util.OptionalLong;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public interface WithRevision<T extends WithId<T>> {
```

Comment:
this seems similar to WithVersion class? Should this be one thing?

Reply:
uh, didn't notice that interface, will have a look

Reply:
So it is not the same thing as WithRevision extends WithId as this is required for db operation, maybe we should support revision for every resource ? (post GA)

Reply:
Perhaps make it WithIdVersioned and have it extend both WithId and WithVersion.  I'd just hate to add both a revision and a version field to an object.

Reply:
I will